### PR TITLE
Add various mechanics tests

### PR DESCRIPTION
**Pull request by DaWoblefet**
Hooray for more tests!
- Uses the FFA hack to support Multi battle tests
- Merges Z-move test files (we had two)
- Adds ~15 new skipped tests (I think), cleans up quite a few more to use the modern format
- Kanban board should have associated cards with the new tests
- Updated as of this post: https://www.smogon.com/forums/posts/9020969/

### CODE REVIEW

File: test/sim/moves/futuresight.js [hunk 1]
```diff
@@ -274,4 +274,63 @@ describe('Future Sight', function () {
...
+		battle.makeChoices();
+		battle.makeChoices('move futuresight', 'auto');
+		battle.makeChoices();
+		battle.makeChoices();
+		const hooh = battle.p2.active[0];
+		const damage = hooh.maxhp - hooh.hp;
+		assert.bounded(damage, [60, 71]); // Damage would be 236-278 if Unaware was being ignored
+	});
+
+	it(`should use the user's most recent current Special Attack stat if the user is on the field`, function () {
```
[the remark this answers is not in the document]
Comment (DaWoblefet) [reply test/sim/moves/futuresight.js]: Added an additional test for stat stage changes.

**Comment by AnnikaCodes** [comment]
@DaWoblefet ?

**Comment by DaWoblefet** [comment]
> @DaWoblefet ?

This is finished, was just giving it a day or two in case anyone had additional comments before merging.

**Comment by AnnikaCodes** [comment]
It's been 13 days; you're good to merge.

**Pull request merged**